aea/contracts/ethereum.py

```diff
@@ -73,17 +73,31 @@ def set_instance(self, ledger_api: EthereumApi) -> None:
         :param ledger_api: the ethereum ledger api
         """
         assert self._instance is None, "Instance already set!"
-        self._instance = ledger_api.api.web3.Contract(
+        self._instance = ledger_api.api.eth.contract(
             abi=self.abi, bytecode=self.bytecode
         )
 
     def set_address(self, ledger_api: EthereumApi, contract_address: str) -> None:
         """
         Set the contract address.
 
+        :param ledger_api: the ledger_api we are using.
         :param contract_address: the contract address
         """
         assert self.instance.address is None, "Address already set!"
-        self._instance = ledger_api.api.web3.Contract(
+        self._instance = ledger_api.api.eth.contract(
+            address=contract_address, abi=self.abi
+        )
+
+    def set_instance_w_address(
+        self, ledger_api: EthereumApi, contract_address: str
+    ) -> None:
+        """
+                Set the contract address.
+
+                :param ledger_api: the ledger_api we are using.
+                :param contract_address: the contract address
+                """
+        self._instance = ledger_api.api.eth.contract(
             address=contract_address, abi=self.abi
         )
```

Comment:
this is identical to set address... I was thinking this should also set the bytecode and have two asserts:
```suggestion
        assert self._instance is None, "Instance already set!"
        assert self.instance.address is None, "Address already set!"
        self._instance = ledger_api.api.eth.contract(
            address=contract_address, abi=self.abi, bytecode=self.bytecode
        )
```

Reply:
What's the point though ? Just not to be identical ?

Reply:
Actually, we don't need line 102, so just 
```suggestion
assert self._instance is None, "Instance already set!"
self._instance = ledger_api.api.eth.contract(
  address=contract_address, abi=self.abi, bytecode=self.bytecode
)
```
that way, we ensure it isn't accidentally called a second time